src/ripple/nodestore/impl/Shard.h

```diff
@@ -88,44 +90,50 @@ class Shard
     void
     sweep();
 
-    bool
-    validate();
-
     std::uint32_t
-    index() const {return index_;}
+    index() const
+    {
+        return index_;
+    }
+
+    std::shared_ptr<Backend> const&
+    getBackend() const;
 
     bool
-    complete() const {assert(backend_); return complete_;}
+    complete() const;
 
-    std::shared_ptr<PCache>&
-    pCache() {assert(backend_); return pCache_;}
+    std::shared_ptr<PCache>
+    pCache() const;
 
-    std::shared_ptr<NCache>&
-    nCache() {assert(backend_); return nCache_;}
+    std::shared_ptr<NCache>
+    nCache() const;
 
     std::uint64_t
-    fileSize() const {assert(backend_); return fileSz_;}
+    fileSize() const;
 
     std::uint32_t
-    fdRequired() const {assert(backend_); return fdRequired_;}
-
-    std::shared_ptr<Backend> const&
-    getBackend() const {assert(backend_); return backend_;}
+    fdRequired() const;
 
     std::shared_ptr<Ledger const>
-    lastStored() {assert(backend_); return lastStored_;}
+    lastStored() const;
+
+    bool
+    validate();
```

Comment:
Just a question... What is the non-`const` thing that `validate()` does?  If all it does is examine the `Shard` and return `true` or `false` about its validity, then it could be `const`.  I suspect that's not all that happens.  At any rate, it seems like `validate()` should either be `const` or have a comment explaining why it is not.

I think, maybe, you can declare `validate() const` if you also declare `valLedger(ledger, next) const` and `valFetch(hash) const`?

Reply:
They can all be const.